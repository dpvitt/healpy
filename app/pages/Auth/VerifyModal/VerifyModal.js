import React from 'react';
import { View, Modal } from 'react-native';
import { Input, Button } from 'react-native-elements';

export const VerifyModal = ({
  modalVisible,
  setConfirmationCode,
  handleConfirmationCode,
}) => (
  <Modal visible={modalVisible}>
    <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}>
      <Input
        label="Confirmation Code"
        leftIcon={{ type: 'font-awesome', name: 'lock' }}
        onChangeText={(value) => setConfirmationCode(value)}
      />
      <Button title="Submit" onPress={handleConfirmationCode} />
    </View>
  </Modal>
);
