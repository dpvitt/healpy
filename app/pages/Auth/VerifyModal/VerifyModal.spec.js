import React from 'react';
import renderer from 'react-test-renderer';
import { VerifyModal } from './VerifyModal';

describe('<VerifyModal />', () => {
  const props = {
    modalVisible: false,
    setConfirmationCode: jest.fn(),
    handleConfirmationCode: jest.fn(),
  };

  const render = renderer.create(<VerifyModal {...props} />);

  it('should render', () => {
    const wrapper = render.toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it('should call setConfirmPassword when confirm password input updated', () => {
    const confirmCodeInput = render.root.findByProps({
      label: 'Confirmation Code',
    });
    confirmCodeInput.props.onChangeText('test');
    expect(props.setConfirmationCode).toHaveBeenCalledWith('test');
  });

  it('should call handleSignIn when button pressed', () => {
    const button = render.root.findByProps({ title: 'Submit' });
    button.props.onPress();
    expect(props.handleConfirmationCode).toHaveBeenCalled();
  });
});
