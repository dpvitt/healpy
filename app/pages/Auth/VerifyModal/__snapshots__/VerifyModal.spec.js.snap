// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VerifyModal /> should render 1`] = `
<Modal
  hardwareAccelerated={false}
  visible={false}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <input
      label="Confirmation Code"
      leftIcon={
        Object {
          "name": "lock",
          "type": "font-awesome",
        }
      }
      onChangeText={[Function]}
    />
    <button
      onPress={[MockFunction]}
      title="Submit"
    />
  </View>
</Modal>
`;
